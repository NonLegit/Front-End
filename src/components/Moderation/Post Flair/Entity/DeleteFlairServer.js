import axios from '../../../../services/instance';
/**
 * Delete Flair
 *
 * @property {string} Name - name of subreddit
 * @property {integer} id - id of flair
 *
 */
const DeleteFlair = async (Name, id) => {
  let statusCode = '';
  await axios.delete(`subreddits/${Name}/flair/${id}`)
    .then((response) => {
      statusCode = response.status;
    }).catch((error) => {
      statusCode = error.response.status;
      console.log(error);
    });
  if (statusCode === 401) {
    window.location.pathname = 'login';
  }
};
export default DeleteFlair;
