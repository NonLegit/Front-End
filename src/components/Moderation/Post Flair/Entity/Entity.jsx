import { IconButton, TableCell } from '@mui/material';
import TableRow from '@mui/material/TableRow';
import DeleteIcon from '@mui/icons-material/Delete';
import { useState } from 'react';
import { useParams } from 'react-router-dom';
import {
  Actions, BodyCell, BodyFirstCell, BodyLastCell, Filter, Text,
} from './style';
import Edit from './EditFlair/Edit';
import DeleteFlair from './DeleteFlairServer';

/**
 * Entity of Flair
 * @component
 * @property  {function} copied copy id
 * @property  {function} Edited show enable form
 * @property  {function} save clase form and enable add
 * @property  {function} cancel clase form and enable add
 * @property  {function} handleClick open form
 * @property  {function} row entity
 *
 * @return {React.Component} - Entity of Flair
 */

function Entity(props) {
  const {
    row, handleClick, can, trueCan,
  } = props;
  const [show, setShow] = useState(false);
  const { subReddit } = useParams();

  function copied() {
    alert('text copied');
  }
  const Edited = () => {
    console.log('clicked');
    handleClick();
    setShow(true);
  };
  const save = () => {
    trueCan();
    setShow(false);
  };
  const cancel = () => {
    trueCan();
    setShow(false);
  };
  return (
    <>
      <TableRow
        key={row?.name}
        sx={{ '&:last-child td, &:last-child th': { border: 0 } }}
      >
        <BodyFirstCell>
          <Text color={row?.textColor} backgroundColor={row?.backgroundColor}>
            {row?.text}
          </Text>
        </BodyFirstCell>
        <BodyCell align="center" />
        <BodyCell align="center" />
        <BodyLastCell align="center">
          <Actions>
            <Filter
              onClick={() => { navigator.clipboard.writeText(row?._id); copied(); }}
            >
              Copy ID
            </Filter>
            {(can)
              ? <Filter onClick={() => Edited()}>edit</Filter>
              : <Filter disabled condition={!can}>edit</Filter>}
            <IconButton onClick={() => { DeleteFlair(subReddit, row._id); }}>
              <DeleteIcon />
            </IconButton>
          </Actions>
        </BodyLastCell>
      </TableRow>
      {show
      && (
      <TableRow>
        <TableCell align="left" colSpan={4}>
          <Edit flair={row} save={save} cancel={cancel} />
        </TableCell>
      </TableRow>
      )}
    </>
  );
}
export default Entity;
