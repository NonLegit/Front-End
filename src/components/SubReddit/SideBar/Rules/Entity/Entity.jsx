import ExpandLessIcon from '@mui/icons-material/ExpandLess';
import ExpandMoreIcon from '@mui/icons-material/ExpandMore';
import { useState } from 'react';
import {
  AboutContent, AboutCountainerRule, AboutDisc, AboutStringRule, Hr,
} from './style';

/**
 * Rule Entity
 * @component
 * @property {object} rule - rule entity
 * @property {integer} index -number of the rule
 * @property  {function} handleClick toggle show rule or not

 * @return {React.Component} - Rule Entity
 */
function Entity(props) {
  const { rule, index } = props;
  const [show, setShow] = useState(false);

  const handleClick = () => {
    setShow(!show);
  };
  return (
    <>
      <AboutCountainerRule key={`${index + 0}`}>
        <AboutStringRule onClick={() => handleClick()}>
          {index + 1}
          .
          {rule.title}
          {show
            ? <ExpandLessIcon />
            : <ExpandMoreIcon />}
        </AboutStringRule>
      </AboutCountainerRule>
      {show
          && (
          <AboutContent>
            <AboutDisc>
              {rule.description}
            </AboutDisc>
          </AboutContent>
          )}
      <Hr />
    </>
  );
}

export default Entity;
