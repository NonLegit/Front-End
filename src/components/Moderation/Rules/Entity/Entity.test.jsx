import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import Entity from './Entity';

// test snapshot
test('test snapshot', async () => {
  const tree = renderer.create(<Entity />).toJSON();
  expect(tree).toMatchSnapshot();
});
