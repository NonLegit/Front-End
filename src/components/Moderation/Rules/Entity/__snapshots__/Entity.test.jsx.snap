// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test snapshot 1`] = `
<div
  className="MuiBox-root css-17y4ujp"
>
  <div
    className="css-u4p24i"
  >
    <div
      className="MuiBox-root css-3lyvs9"
    >
      NaN
    </div>
    <div
      className="MuiBox-root css-7dy4od"
    />
  </div>
  <div
    className="css-u4p24i"
  >
    <button
      className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="CreateIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 0 0-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
        />
      </svg>
    </button>
    <div
      className="MuiBox-root css-orlaaz"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1or9raf-MuiSvgIcon-root"
        data-testid="UnfoldMoreOutlinedIcon"
        focusable="false"
        onClick={[Function]}
        viewBox="0 0 24 24"
      >
        <path
          d="M12 5.83 15.17 9l1.41-1.41L12 3 7.41 7.59 8.83 9 12 5.83zm0 12.34L8.83 15l-1.41 1.41L12 21l4.59-4.59L15.17 15 12 18.17z"
        />
      </svg>
    </div>
  </div>
</div>
`;
